hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -296,13 +296,18 @@ private long acquireLock() throws UnknownHostException, TException, InterruptedE
   private void unlock(Optional<Long> lockId) {
     if (lockId.isPresent()) {
       try {
-        metaClients.run(client -> {
-          client.unlock(lockId.get());
-          return null;
-        });
+        doUnlock(lockId.get());
       } catch (Exception e) {
-        throw new RuntimeException(String.format("Failed to unlock %s.%s", database, tableName), e);
+        LOG.warn("Failed to unlock {}.{}", database, tableName, e);
```

Comment:
I'm generally ok, but what's the behavior you've seen on the server side? Can new locks still be acquired by other clients or the table is left in a locked state forever?

Reply:
If an update succeeds, then this method _must_ return success.

If it doesn't return successfully but did update, then the caller will delete the metadata file that was attempted. But that metadata file was actually commited. In that case, the table is broken.

Also note that Iceberg _must_ delete the metadata file that was attempted if the commit fails. If not, then the situation where an attempt succeeds but the method throws an exception can result in a retry, which could succeed because the table is fine. In that case, the table has silently committed the same data twice. It is better to break the table than to corrupt the data.